src/plugins/pass/main.js

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/gjs
+
+const { GLib, Gio, Notify } = imports.gi;
+
+Notify.init('Password Store');
+
+/** The directory that this script is executed from. */
+const SCRIPT_DIR = GLib.path_get_dirname(new Error().stack.split(':')[0].slice(1));
+
+/** Add our directory so we can import modules from it. */
+imports.searchPath.push(SCRIPT_DIR)
+
+const STDIN = new Gio.DataInputStream({ base_stream: new Gio.UnixInputStream({ fd: 0 }) })
+const STDOUT = new Gio.DataOutputStream({ base_stream: new Gio.UnixOutputStream({ fd: 1 }) })
+
+
+function list_passwords(root, directory) {
+    const entries = directory.enumerate_children(
+        Gio.FILE_ATTRIBUTE_STANDARD_NAME + "," +
+        Gio.FILE_ATTRIBUTE_STANDARD_TYPE + "," +
+        Gio.FILE_ATTRIBUTE_STANDARD_IS_HIDDEN,
+        Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS,
+        null
+    );
+
+    let entry;
+    let results = [];
+
+    while((entry = entries.next_file(null)) !== null) {
+        const name = entry.get_name();
+        const target_file = directory.get_child(entry.get_name());
+
+        if (entry.get_file_type() === Gio.FileType.DIRECTORY) {
+            results = results.concat(list_passwords(root, target_file))
+        } else if (name.endsWith('.gpg')) {
+            let path = root.get_relative_path(target_file);
+            results.push(path.replace('.gpg', ''));
+        }
+    }
+    return results;
+}
+
+function send_notification(body, icon) {
+    let notification = Notify.Notification.new(
+        'Password Store',
+        body,
+        icon,
+    );
+    notification.show();
+}
+
+
+class App {
+    constructor() {
+        let home_dir = GLib.get_home_dir();
+        let password_store = Gio.File.new_for_path(`${home_dir}/.password-store`);
```

Comment:
shouldn't this first try to read PASSWORD_STORE_DIR environment variable and fallback to this value?

Reply:
To be honest - I was not aware this environment variable option existed. Do you have any links to documentation for this? I couldnt find it on https://www.passwordstore.org/

Reply:
You can see it here for example https://wiki.archlinux.org/index.php/Pass#Advanced_usage or you can look at `man pass` and then search for "ENVIRONMENT VARIABLES" there is a list of all supported options (the first one being the store dir.